### ISSUE TITLE
[Telemetry] Design shared logging module

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/pocket#164]
# Objective
Create the foundations to unify and standardize the logging approach across the entire codebase.

# Origin Document

See the github discussion for full details: https://github.com/pokt-network/pocket/discussions/214

The implementation of this was started by @derrandz here: https://github.com/pokt-network/pocket/pull/162.

This ticket can either build on top of the existing WIP or use those learnings and apply them to a different approach.

# Goals / Deliverables
- [x] Research and identify common logging frameworks available in Go and use in major codebases (e.g. Uber)
- [ ] Design a new shared logging module based on the results from the research above. This means it could be:
    - Created from scratch
    - Imports an existing framework
    - Extends an existing one with a lightweight wrapper

## General issue checklist
- [ ] Update the appropriate CHANGELOG
- [ ] Update the appropriate README on how to use and test this feature
- [ ] Update any relevant global documentation & references
- [ ] Add a video and/or gif on how to use feature
- [ ] Document small issues / TODOs along the way

# Non-goals
- Add new logs throughout the codebase
- Improve the explicitness or correctness of existing logs
- Add new business logic functionality

# Testing Methodology
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)
Remove

---

Creator: @Olshansk 
Co-Owners: @vlad-she @derrandz

**Comment by Olshansk** [comment]
Note that there will be cross-dependencies between this and https://github.com/pokt-network/pocket/issues/118

**Comment by Olshansk** [comment]
A WIP implementation of this is available here: https://github.com/pokt-network/pocket/pull/162

**Comment by Olshansk** [comment]
THis is blocked by https://github.com/pokt-network/pocket/issues/181

**Comment by jessicadaugherty** [comment]
Split by
- Design
- Implementation/support

Remove
- Grafana (cover in #189)

**Comment by okdas** [comment]
I started working on this, will post a PR soon.
